src/poetry/utils/requirements.py

```diff
@@ -0,0 +1,133 @@
+
+    try:
+        cwd = command.poetry.file.parent
+    except (PyProjectException, RuntimeError):
+        cwd = Path.cwd()
+
+    return [
+        parse_dependency_specification(
+            requirement=requirement,
+            env=env,
+            cwd=cwd,
+        )
+        for requirement in requirements
+    ]
+
+
+def format_requirements(requirements: list[dict[str, str]]) -> Requirements:
+    requires: Requirements = {}
+    for requirement in requirements:
+        name = requirement.pop("name")
+        constraint: str | InlineTable
+        if "version" in requirement and len(requirement) == 1:
+            constraint = requirement["version"]
+        else:
+            constraint = inline_table()
+            constraint.trivia.trail = "\n"
+            constraint.update(requirement)
+
+        requires[name] = constraint
+
+    return requires
+
+
+def find_best_version_for_package(
+    pool: Pool,
+    name: str,
+    required_version: str | None = None,
+    allow_prereleases: bool = False,
+    source: str | None = None,
+) -> tuple[str, str]:
+    from poetry.version.version_selector import VersionSelector
+
+    selector = VersionSelector(pool)
+    package = selector.find_best_candidate(
+        name, required_version, allow_prereleases=allow_prereleases, source=source
+    )
+
+    if not package:
+        # TODO: find similar
+        raise ValueError(f"Could not find a matching version of package {name}")
+
+    return package.pretty_name, selector.find_recommended_require_version(package)
+
+
+def determine_requirements_from_list(
+    command: Command,
+    pool: Pool,
+    requires: list[str],
+    allow_prereleases: bool = False,
+    source: str | None = None,
+) -> list[dict[str, Any]]:
+    result = []
+    for requirement in parse_requirements(requires, command, None):
+        if "git" in requirement or "url" in requirement or "path" in requirement:
+            result.append(requirement)
+            continue
+        elif "version" not in requirement:
+            # determine the best version automatically
+            name, version = find_best_version_for_package(
+                pool,
+                requirement["name"],
+                allow_prereleases=allow_prereleases,
+                source=source,
+            )
+            requirement["version"] = version
+            requirement["name"] = name
+
+            command.line(f"Using version <b>{version}</b> for <c1>{name}</c1>")
+        else:
+            # check that the specified version/constraint exists
+            # before we proceed
+            name, _ = find_best_version_for_package(
+                pool,
+                requirement["name"],
+                requirement["version"],
+                allow_prereleases=allow_prereleases,
+                source=source,
+            )
+
+            requirement["name"] = name
+
+        result.append(requirement)
+
+    return result
+
+
+def create_pool() -> Pool:
```

Comment:
Let's drop this helper and keep things inline -- it's less messy than stashing this helper in such an obscure place, and this isn't the PR to solve the ergonomics of creating the pool in anyway. If you want to clean things up and be a bit more specific, you can just use `PyPiRepository.search()` instead of creating a pool with one repository.

Reply:
> If you want to clean things up and be a bit more specific, you can just use PyPiRepository.search() instead of creating a pool with one repository.

I'm all for cleaning things up, but not sure what it means to replace this with `PyPiRepository.search()` -> moving the code inline. Also, good to avoid additional refactoring in this PR's scope.